src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -321,61 +315,33 @@ internal class ProjectPlugin(private val project: Project) {
         return@afterEvaluate
       }
 
-      val testSource = if (shouldAnalyzeTests()) sourceSets.findByName(SourceSet.TEST_SOURCE_SET_NAME) else null
-      val mainSource = sourceSets.findByName(SourceSet.MAIN_SOURCE_SET_NAME)
-      mainSource?.let { sourceSet ->
-        try {
-          // Regardless of the fact that this is a "java-library" project, the presence of Spring
-          // Boot indicates an app project.
-          val dependencyAnalyzer = if (pluginManager.hasPlugin(SPRING_BOOT_PLUGIN)) {
-            logger.warn(
-              "(dependency analysis) You have both java-library and org.springframework.boot applied. You probably " +
-                "want java, not java-library."
-            )
-            JavaAppAnalyzer(
-              project = this,
-              sourceSet = sourceSet,
-              testSourceSet = testSource,
-              kind = SourceSetKind.MAIN
-            )
-          } else {
-            JavaLibAnalyzer(
-              project = this,
-              sourceSet = sourceSet,
-              testSourceSet = testSource,
-              kind = SourceSetKind.MAIN,
-              hasAbi = true
-            )
-          }
-          analyzeDependencies(dependencyAnalyzer)
-        } catch (_: UnknownTaskException) {
-          logger.warn("Skipping tasks creation for sourceSet `${sourceSet.name}`")
-        }
-      }
-
-      testSource?.let { sourceSet ->
-        try {
-          // Regardless of the fact that this is a "java-library" project, the presence of Spring
-          // Boot indicates an app project.
-          val dependencyAnalyzer = if (pluginManager.hasPlugin(SPRING_BOOT_PLUGIN)) {
-            JavaAppAnalyzer(
-              project = this,
-              sourceSet = sourceSet,
-              testSourceSet = null,
-              kind = SourceSetKind.TEST
-            )
-          } else {
-            JavaLibAnalyzer(
-              project = this,
-              sourceSet = sourceSet,
-              testSourceSet = null,
-              kind = SourceSetKind.TEST,
-              hasAbi = false
-            )
+      sourceSets.forEach { sourceSet ->
```

Comment:
When we work with the "Source Set Name" instead of a "SourceSetType", for plain Java/Kotlin we can iterate all source sets.